pydis_site/apps/api/viewsets/bot/nomination.py

```diff
@@ -1,32 +1,224 @@
+    Create a new, active nomination returns the created nominations.
+    The `user`, `reason` and `actor` fields are required and the `user`
+    and `actor` need to know by the site. Providing other valid fields
+    is not allowed and invalid fields are ignored. A `user` is only
+    allowed one active nomination at a time.
+
+    #### Request body
+    >>> {
+    ...     'actor': 409107086526644234
+    ...     'reason': 'He would make a great helper',
+    ...     'user': 409107086526644234
+    ... }
+
+    ### Response format
+    See `GET /bot/nominations/<id:int>`
 
+    #### Status codes
+    - 201: returned on success
+    - 400: returned on failure for one of the following reasons:
+        - A user already has an active nomination;
+        - The `user` or `actor` are unknown to the site;
+        - The request contained a field that cannot be set at creation.
+
+    ### PATCH /bot/nominations/<id:int>
+    Update the nomination with the given `id` and return the updated nomination.
+    For active nominations, only the `reason` may be updated; for inactive
+    nominations, both the `reason` and `unnominate_reason` may be updated.
+
+    #### Request body
+    >>> {
+    ...     'reason': 'He would make a great helper',
+    ...     'unnominate_reason': 'He needs some time to mature his Python knowledge'
+    ... }
+
+    ### Response format
+    See `GET /bot/nominations/<id:int>`
+
+    ## Status codes
+    - 200: returned on success
+    - 400: if a field in the request body is invalid or disallowed
+    - 404: if an infraction with the given `id` could not be found
+
+    ### PATCH /bot/nominations/<id:int>/end_nomination
+    Ends an active nomination and returns the updated nomination.
+
+    The `unnominate_reason` field is the only allowed and required field
+    for this operation. The nomination will automatically be marked as
+    `active = false` and the datetime of this operation will be added to
+    the `unwatched_at` field.
+
+    #### Request body
+    >>> {
+    ...     'unnominate_reason': 'He needs some time to mature his Python knowledge'
+    ... }
+
+    ### Response format
+    See `GET /bot/nominations/<id:int>`
+
+    #### Status codes
+    - 200: returned on success
+    - 400: returned on failure for the following reasons:
+        - `unnominate_reason` is missing from the request body;
+        - Any other field is present in the request body;
+        - The nomination was already inactiive.
+    - 404: if an infraction with the given `id` could not be found
+    """
     serializer_class = NominationSerializer
-    queryset = Nomination.objects.prefetch_related('author', 'user')
-    frozen_fields = ('author', 'inserted_at', 'user')
+    queryset = Nomination.objects.all()
+    filter_backends = (DjangoFilterBackend, SearchFilter, OrderingFilter)
+    filter_fields = ('user__id', 'actor__id', 'active')
+    frozen_fields = ('id', 'actor', 'inserted_at', 'user', 'unwatched_at', 'active')
```

Comment:
I'm not sure, I followed the format/style of some of the other API endpoints that also use a `frozen_fields` check like that. This is my first time doing anything in Django and DRF, so I'll read the docs and experiment for a bit.

Reply:
I think this would be an interesting thing to rewrite in the future, but since it doesn't change the way the API works from the outside or the model, I think we should look into this at a later time. I'd like to focus on the other things that need to be done first, like the getting the django branch of the bot ready.

Reply:
Fair enough. Plus, this is apparently consistent with other view sets so any possible changes to how this is done can be batched together in the future.